### ISSUE TITLE
Decide how to handle the terminal in a dfhooks world

### ISSUE THREAD

**Issue by myk002** [DFHack/dfhack#2935]
Can we still hook stdin and stdout to interface with the host terminal, or does the init hook come too late?

What are our options (aside from my having a terminal)?

**Comment by lethosor** [comment]
The problem I was referring to on Discord is that there __is no__ host terminal if the DF executable is launched directly on Linux/macOS.

Another option I brought up is developing our own console window with SDL2, but that is quite a large undertaking to achieve parity with a native terminal.

**Comment by myk002** [comment]
another option with SDL2 is to populate the environment variable that essentially allows us to do what we do now -- hook SDL functions at DLL load time (the name of this variable escapes me right now, and I can't seem to find the list of SDL2 env vars online for some reason, but I'm quite sure it's there). that would still allow us to avoid overwriting SDL.dll. It's not exactly the same as what we do now, though. Would that still let us set up stdin and stdout?

**Comment by myk002** [comment]
another option is to implement a shell in dfhack-run. we'd start up without a terminal, but dfhack-run would become the terminal (a la ssh)

**Comment by lethosor** [comment]
> Would that still let us set up stdin and stdout?

No, SDL2 is orthogonal to the terminal issue on *nix - DF can be launched with or without a dedicated terminal, regardless of whether it uses SDL2.

**Comment by myk002** [comment]
Option for providing our own dynamic console: https://github.com/rlt3/SDL-Console
